### DISCUSSION TITLE
RaspberryMatic 3.69.7.20230626

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2348] [Announcements]
This is release 3.69.7.20230626 of RaspberryMatic which is a **maintenance** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.69.7.20230626/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.69.7.20230626) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=79219)</sub>

## :construction: Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.69.7.20230506...3.69.7.20230626).</sub>

#### CCU/HomeMatic service changes:
- updated [OCCU](https://github.com/eq-3/occu) version to [3.69.7-3](https://github.com/jens-maus/occu/tree/b_3_69) which comes with the following changes:
   - updated `ReGaHss` logic engine to version `R1.00.0388.0235 (Jun 13 2023)`:
      - added new datapoint applicable `LastChangeValue()`, `LastChangeTimestamp()` and `LastChangeTimestampSeconds()` functions to query the value and timestamps of the last changed value a datapoint had before a new value state (cf. https://homematic-forum.de/forum/viewtopic.php?f=65&t=79126).
- fixed missing `RPI-RF-MOD` blue led blinking during network setup phase on startup.
- improved `RPI-RF-MOD` led handling in LAN-Gateway mode (HMLGW) so that the LED will be properly set to blue once the system has booted and also make sure not to start `hss_led` at all.
- modified certain unnecessary cron-based maintenance job executions or watchdog tools like `triggerAlarm.tcl` to be skipped if system is running in LAN-Gateway mode (HMLGW).
- when resetting a `raw-uart` it can happen that a certain rf module type (`HmIP-RFUSB`) does not support that procedure and returns an error. However, we need to simply ignore that, thus dump that error to null.
- allow the `checkFirmwareUpdate.sh` script for update check but output an error in case it is used on the docker/oci platform for trying to apply an update right away (which is not possible).
- introduced new `install-docker.sh` docker/oci installation script to setup a docker-based RaspberryMatic with a `macvlan` based network connection for allowing to connect a `HmIP-HAP` or `HmIPW-DRAP` which requires multicast udp traffic for proper communication (#1373, https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI).
- modified the HomeAssistant Add-on documentation to explain a temporary workaround to get a `HmIP-HAP` or `HmIPW-DRAP` connected to a running RaspberryMatic HA add-on by a manual `macvlan` network setup until the HA developers have implemented the required changes that a HA add-on can setup a `macvlan`-based network connection itself (cf. https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#hmip-haphmipw-drap-support-patch).
- updated `detect-radio-module` tool and `generic_raw_uart` rf module kernel driver to latest versions available.
- fixed issue in LAN-Gateway daemon (`hmlangw`) daemon where during bootloader setup the daemon could end up in an endless loop never starting and outputting massive `eq3loop_write_master()` kernel errors due to serial data flooding. To fix this issue we now check for a different bootloader reply scheme after sending the bootloader entering sequence (#2340).
 
#### WebUI changes:
- modified the help.cgi patch to also output the network configuration settings on the general help page.
- updated CodeMirror to latest 5.65.13
- integrated improved png and svg-based RaspberryMatic logo files (#2341, @Rygle).

#### Operating system changes:
- updated upstream kernel to 6.1.35.
- updated buildroot monit upstream patch to use latest 5.33.0 version.
- updated buildroot to latest 2023.05.
- updated rpi-eeprom to latest pieeprom-2023-05-11.bin version.
- removed rng-tools because in newer linux kernel versions `/dev/random` is not blocking anymore (5.6+) and the standard kernel provided entropy pool is strong enough (5.10+) so that tools like rng-tools or haveged are not required anymore and just unnecessarily consume CPU time. (cf. https://github.com/nhorman/rng-tools/issues/195#issuecomment-1520598021, https://forum.manjaro.org/t/low-entropy-on-my-system/119233).
- updated tailscale to latest 1.44.0.
- integrated upstream busybox patch for fixing an illegal instruction on rare CPU constellations when executing sha256sum. (cf. #2309, https://bugs.busybox.net/show_bug.cgi?id=15236).
- renamed `S03InitURandom` startup script to `S03seedrng` and reworked it to use the new `seedrng` busybox tool to init and seed the kernel RNG rather than manually doing this including the `rndaddtoentcnt` tool. Thus, also retired this tool in favor of the new seedrng busybox tool.

## :family: Contributors (alphabetically):
- @jens-maus
- @Rygle

## :memo: Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## :package:  Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-rpi4.zip)<br/>SHA256: cfd484d29094c8ef5fa10f8a658118235a959a50b27256779f206c8ffdbc241f</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-rpi3.zip)<br/>SHA256: e30eb1f198ce989c558f650e2148cf3736301fa7f8d9ace3873246ce3ab27a73</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: f9ab8de6bdb7a935a29c5c01265f5922d95d442aabe4fb185157ac4bf5950389</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-rpi2.zip)<br/>SHA256: 26760efc1095cea191a735413907fd6cbfdc953323dc03c09355ccaa8286b335</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-rpi0.zip)<br/>SHA256: f1d346ef672b01775fc43c6167070262fb03fc20c76f4817ef37eb8eed855736</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-tinkerboard.zip)<br/>SHA256: 6c8c3d02c60f07b5fc0f28c2ad7ccf62a3b163d25656cd1424f1444d35cb34cf</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-odroid-n2.zip)<br/>SHA256: 29e86a5a3f88f6324f407cc6b4a7b9f83d3cf520d8b229fa93c76f1ad79c04ad</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-odroid-c4.zip)<br/>SHA256: a80b1b86150df24f0a73dffd46e3ac219030fe700f0fd8b254524ff074dbd005</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-odroid-c2.zip)<br/>SHA256: bd3ca4d957ec37a4d06007dfa2e2695dbbb4fc278e5e27ae67b4a4a22184e0f1</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-intelnuc.zip)<br/>SHA256: ad1cd5a0c8ce9401e624cf32c1d5801abfb583da9b70cbfd2c17712738a8acdf</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-ova.zip)<br/>SHA256: 4adf25ae893af340e367dec96711f4dec1ae9ac1c52f6b7fcc3c8b5b936507ac</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626.ova) (only for initial OVA installation)<br/>SHA256: 669ddfc2b6ebc5ef33c25bed2ad10bab2e70094b86f66a98c58ffe8a29a2d4bf</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>:package: [RaspberryMatic-3.69.7.20230626-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: c0a28fa44ccfded61183c0e3013460acfa60fb1569411ee363fab70c6dcd7543</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: 83e009cc4e1431d7a0effa2cfb9de02231025251aea1baa70129dd1fd9abf5a9</sub>
<sub>:package: [RaspberryMatic-3.69.7.20230626-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.69.7.20230626/RaspberryMatic-3.69.7.20230626-oci_arm.tgz) (arm/armhf)<br/>SHA256: 94583b2e49527cd959cba66a824ecbd7f69fee97cbfcceddf98818bd71c832e4</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.69.7.20230626'>RaspberryMatic 3.69.7.20230626</a>.</em>

**Comment by Weseler485** [comment]
Der Link zum deutschen Beitrag funktioniert nicht

**Comment by kornet9m133** [comment]
when booting the vm created with the .ova file I see the dhcp network address and some error messages like 
can't create directory '/etc/local/crontabs' No space left on device
so the boot process seams to not finish, and I can't connect to the gui
I can't login with ssh either because I don't get into the gui first.

what is the intended use of the ..-ova.img file ? can anyone please help ?
thank you

**Reply by jens-maus** [reply]
This sounds like you forgot to enlarge the virtual HDD during your ova installation steps. So please walk through the official ova installation steps very closely...

**Reply by kornet9m133** [reply]
YES, i made a mistake in the virtual media manager. now it works fine, thank you so mutch
